From cb68870dea669e6c7ae2dd7774b32ef873eac637 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cote2004-github@yahoo.com>
Date: Thu, 7 Feb 2019 09:57:55 -0200
Subject: [PATCH 2/2] crypto.h: add afalg engine to builtin list

Signed-off-by: Eneas U de Queiroz <cote2004-github@yahoo.com>

diff --git a/include/openssl/crypto.h b/include/openssl/crypto.h
index 7e50b1bf46..aa503e8e39 100644
--- a/include/openssl/crypto.h
+++ b/include/openssl/crypto.h
@@ -384,7 +384,7 @@ int CRYPTO_memcmp(const void * in_a, const void * in_b, size_t len);
 # define OPENSSL_INIT_ENGINE_ALL_BUILTIN \
     (OPENSSL_INIT_ENGINE_RDRAND | OPENSSL_INIT_ENGINE_DYNAMIC \
     | OPENSSL_INIT_ENGINE_CRYPTODEV | OPENSSL_INIT_ENGINE_CAPI | \
-    OPENSSL_INIT_ENGINE_PADLOCK)
+    OPENSSL_INIT_ENGINE_PADLOCK | OPENSSL_INIT_ENGINE_AFALG)
 
 
 /* Library initialisation functions */
